### PR TITLE
.github/workflows: Added initial build support based on github actions

### PR DESCRIPTION
**Pull request by mphalke**
Created build .yaml file to build all the projects

### CODE REVIEW

**Review by jansunil — APPROVED** [review]

**Comment by ribdp** [comment]
> If change is detected in any particular project folder, the build is triggered for that project

Is this implemented in the yaml file?

**Comment by mphalke** [comment]
> > If change is detected in any particular project folder, the build is triggered for that project
> 
> Is this implemented in the yaml file?

No, updated the comment now. Need to work on that yet

**Comment by CMinajigi** [comment]
is this building projects now? How can we add matrix combination here.

**Comment by mphalke** [comment]
> is this building projects now? How can we add matrix combination here.



> is this building projects now? How can we add matrix combination here.

Yes all projects are building with default matrix combinations. Adding project based matrix combination is not in the scope of this PR and need to work on that in near future

**Review by CMinajigi — APPROVED** [review]

File: .github/workflows/build_projects.yaml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+          pushd projects/${{ matrix.project }}
+          make NEW_CFLAGS+=-DPLATFORM_NAME=${{ matrix.platform }}
+          popd
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ matrix.project }}
+          path: |
+            projects/${{ matrix.project }}/build/${{ matrix.project }}.bin
```
[the remark this answers is not in the document]
Comment (mphalke) [reply .github/workflows/build_projects.yaml:70]: [[github/workflows: Added initial build support based on github actions · analogdevicesinc/precision-converters-firmware@b57e117](https://github.com/analogdevicesinc/precision-converters-firmware/actions/runs/6544882173)](https://github.com/analogdevicesinc/precision-converters-firmware/actions/runs/6547137835)

**Comment by mphalke** [comment]
Hi all, any other comments?

**Review by jansunil — APPROVED** [review]

**Review by SaikiranGudla — APPROVED** [review]

**Review by ribdp — APPROVED** [review]

**Pull request merged**